service/frontend/workflowHandler.go

```diff
@@ -1113,8 +1117,9 @@ func (wh *WorkflowHandler) PollActivityTaskQueue(ctx context.Context, request *w
 		}
 	}
 
+	nilResponse := &workflowservice.PollActivityTaskQueueResponse{}
 	if matchingResponse == nil {
-		return nil, nil
+		return nilResponse, nil
 	}
```

Comment:
I think this entire check needs to be removed. If `err ==nil` (we get here only in this case) `matchingResponse` should always be non-nil.